lib/services/page-service.ts

```diff
@@ -71,6 +85,93 @@ export class PageService implements IPageService {
                 });
         });
     }
+
+    private createPage(srcDir: string, destDir: string, pageName: string): Promise<string> {
+        const camelCaseNameString = _.camelCase(pageName);
+        const pageDirPath = util.path.join(destDir, pageName);
+
+        const newPageName = {
+            originalName: pageName,
+            pascalCaseName: _.upperFirst(camelCaseNameString),
+            camelCaseName: camelCaseNameString
+        };
+
+        return new Promise((resolve, reject) => {
+            this.prepareRenderedFiles(srcDir, newPageName)
+                .then((renderedPageFiles) => {
+                    util.fs.ensureDir(pageDirPath)
+                        .then(() => {
+                            renderedPageFiles.forEach((page: any) => {
+                                const filePath = util.path.join(pageDirPath, page.filename);
+
+                                try {
+                                    util.fs.writeFileSync(filePath, page.content, "utf8");
+                                } catch (error) {
+                                    return Promise.reject(error);
+                                }
+                            });
+
+                            return resolve();
+                        });
+                }).catch((error) => {
+                    reject(error);
+                });
+        });
+    }
+
+    private renderFile(filePath: string, pageName: any): Promise<any> {
+        const directoryName = util.path.basename(util.path.dirname(filePath));
+        const filename = util.path.basename(filePath);
+        const newFileName = filename.replace(directoryName, pageName.originalName);
+
+        const data = {
+            OriginalName: pageName.originalName,
+            PascalCaseName: pageName.pascalCaseName,
+            CamelCaseName: pageName.camelCaseName
+        };
+
+        return new Promise((resolve, reject) => {
+            ejs.renderFile(filePath, data, (error: any, renderedContent: any) => {
+                if (error) {
+                    return Promise.reject(new Error("Fail to render file with: " + error));
+                } else {
+                    resolve({
+                        filename: newFileName,
+                        content: renderedContent
+                    });
+                }
+            });
+        });
+    }
+
+    private prepareRenderedFiles(srcDir: string, pageName: any): Promise<Array<any>> {
+        return new Promise((resolve, reject) => {
+            util.fs.readdir(srcDir, (error: any, files: any) => {
+                if (error) {
+                    return Promise.reject(error);
```

Comment:
I think this should be simply 'reject' since we are not in a promise chain here.